libs/utils/utils.js

```diff
@@ -647,21 +647,27 @@ const decorateCopyLink = (a, evt) => {
 };
 
 export function convertStageLinks({ anchors, config, hostname, href }) {
```

Comment:
I lack the context, hence the question:
No matter if this is done _before_ or _after_ the localization, we'd just swap out the host, no? Since localization is part of the path - we aren't touching it and this should already be done somewhere from the milo links conversion 🤔

Reply:
I agree, this does look a bit more convoluted than needed, at least at a first glance. The `localizeLink` method should handle all the complexities of handling the locale prefix. There are use-cases we need to consider, such as `#_dnt` links, which should not be translated. That even has a `originHostName` parameter, which might come in handy when trying to switch up origins.

Reply:
These changes were necessary for use cases where different consumers share the same domain but are differentiated by path (e.g., bacom and bacom blog).

For example, without the updates in this PR, if a consumer sets the following mapping:
```
  '^https://some.consumer': {
    '^https://business.adobe.com(?!/blog)': 'https://business.stage.adobe.com',
    '^https://business.adobe.com/blog': 'https://main--bacom-blog--adobecom.hlx.page',
  },

```
they would expect that, on `https://some.consumer`, `business.adobe.com` would convert to `business.stage.adobe.com` and `business.adobe.com/blog` to `main--bacom-blog--adobecom.hlx.page`.

This works when localization isn’t involved, but on `https://some.consumer/de`, both `business.adobe.com/blog` and `business.adobe.com` would convert to https://business.stage.adobe.com.”, which is wrong.

My solution is to remove the localization from the path, do the conversions and then reattach the localization.

Reply:
That sounds a little confusing, would it not be already possible to add the locale part to the regex as an optional parameter? 🤔

Reply:
That was my initial thought, but you'll end up losing the locale part when doing the conversion with String.replace()

Reply:
@robert-bogos I feel super bad for writing this now, but still curious what you think. What if we made this a config option instead? Every consumer can decide for themselves if they want their stage links converted with an option like
`stageLinkTransformation: on`

Then this could be opt in, you'd have way simpler code code, we'd be sending less bytes to consumers. What do you think?

Reply:
cc: @mokimo @narcis-radu @overmyheadandbody if you want to chime in too :)

Reply:
@vhargrave this is already opt in, albeit the config is a bit more involved to get this running rather than just `on`. We had discussed things back and forth a lot, but settled on this and had a few iterations of it.

This feature is not as trivial as one might think 😬

Reply:
I see, thanks for explaining and sending this [discussion](https://github.com/orgs/adobecom/discussions/2880) @mokimo 🙌 
I didn't realize the requirements were so broad and that config option certainly won't cover it lol.